### PR TITLE
Use placeholder templates for registration params

### PR DESCRIPTION
**Pull request by katrogan**

### CODE REVIEW

File: flytekit/clis/helpers.py [hunk 1]
```diff
@@ -89,9 +89,14 @@ def str2bool(str):
 def _hydrate_identifier(
     project: str, domain: str, version: str, identifier: _identifier_pb2.Identifier
 ) -> _identifier_pb2.Identifier:
-    identifier.project = identifier.project or project
-    identifier.domain = identifier.domain or domain
-    identifier.version = identifier.version or version
+    if not identifier.project or identifier.project == "{{ .project }}":
```
Comment (kumare3) [inline flytekit/clis/helpers.py:92]: should these not be regexes?

Comment (kumare3) [reply flytekit/clis/helpers.py:92]: if you want example regex look at https://github.com/lyft/flytekit/blob/annotations/flytekit/annotated/base_sql_task.py#L15

Comment (katrogan) [reply flytekit/clis/helpers.py:92]: perhaps, but in the case of the project field the only templatized variable that can ever be used is project

**Comment by wild-endeavor** [comment]
auth role and raw output data prefix?

**Comment by wild-endeavor** [comment]
Also should it just be .project?  Or should we bucket them under registration.project or something?  @katrogan @kumare3

**Comment by katrogan** [comment]
> auth role and raw output data prefix?

we use the config values at serialization time.

**Comment by katrogan** [comment]
no preference on `{{ .project }}` vs `{{ registration.project }}`

**Comment by kumare3** [comment]
I like the idea of `registration.project` thus registration is a protected namespace

**Comment by kumare3** [comment]
> > auth role and raw output data prefix?
> 
> we use the config values at serialization time.

they shouldnt be at serialization right? actually tbh, they are just Admin attributes so should probably do not even exist at serialization time?

**Comment by katrogan** [comment]
@kumare3 yep eventually the goal to make admin handle them https://github.com/lyft/flytectl/pull/18

i'll update

**Comment by katrogan** [comment]
PTAL

File: flytekit/clis/helpers.py (hunk 1 above)
Comment (kumare3) [reply flytekit/clis/helpers.py:92]: ya but what about whitespaces?

Comment (kumare3) [reply flytekit/clis/helpers.py:92]: also this should be changed right?

Comment (katrogan) [reply flytekit/clis/helpers.py]: derp, yes. had my test case backwards 🤦

Comment (katrogan) [reply flytekit/clis/helpers.py]: re: whitespace, the templating only ever comes from flytesdk right? presumably we can trust that to properly assign the constants, whitespace and all?

File: flytekit/clis/flyte_cli/main.py [hunk 2]
```diff
@@ -418,7 +419,9 @@ def _render_schedule_expr(lp):
     "--assumable-iam-role", help="Custom assumable iam auth role to register launch plans with"
 )
 _kubernetes_service_acct_option = _click.option(
-    "--kubernetes-service-account", help="Custom kubernetes service account auth role to register launch plans with"
+    "-s",
+    "--kubernetes-service-account",
+    help="Custom kubernetes service account auth role to register launch plans with",
 )
 _output_location_prefix_option = _click.option(
     "--output-location-prefix", help="Custom output location prefix for offloaded types (files/schemas)"
```
Comment (wild-endeavor) [inline flytekit/clis/flyte_cli/main.py:427]: Can we add a short form for this too?

Comment (katrogan) [reply flytekit/clis/flyte_cli/main.py]: ok. done

File: flytekit/clis/sdk_in_container/serialize.py [hunk 3]
```diff
@@ -18,15 +18,14 @@
...
-from flytekit.configuration import auth as _auth_config
 from flytekit.configuration import internal as _internal_config
 from flytekit.tools.fast_registration import compute_digest as _compute_digest
 from flytekit.tools.fast_registration import filter_tar_file_fn as _filter_tar_file_fn
 from flytekit.tools.module_loader import iterate_registerable_entities_in_order
 
-_PROJECT_PLACEHOLDER = ""
-_DOMAIN_PLACEHOLDER = ""
-_VERSION_PLACEHOLDER = ""
+_PROJECT_PLACEHOLDER = "{{ registration.project }}"
```
Comment (wild-endeavor) [inline flytekit/clis/sdk_in_container/serialize.py:26]: Can you summarize why there aren't placeholders for auth and output prefix?

Comment (katrogan) [reply flytekit/clis/sdk_in_container/serialize.py:26]: neither are required during serialization

besides auth is a oneof - what should i have a placeholder for: the k8s service account or assumable iam role?

**Comment by katrogan** [comment]
friendly ping

**Comment by katrogan** [comment]
ping @wild-endeavor @kumare3

**Comment by kumare3** [comment]
lgtm, I will wait for @wild-endeavor to +1

File: flytekit/clis/sdk_in_container/serialize.py (hunk 3 above)
Comment (katrogan) [reply flytekit/clis/sdk_in_container/serialize.py:29]: done

**Comment by katrogan** [comment]
approved offline
